### PR TITLE
Fix Development Server Warnings and Enhance Project Configuration

### PR DESCRIPTION
**Pull request by snipher-marube**
**Description:**

This pull request addresses and resolves several warnings encountered during the development server startup. The changes made improve the overall configuration of the project to ensure a smoother development experience and adherence to Django’s best practices.

**Changes Made:**

1. **Configured DEFAULT_AUTO_FIELD Setting:**

-  _Warning fixed_
```
HINT: Configure the DEFAULT_AUTO_FIELD setting or the AppConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
registration.user_type: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
HINT: Configure the DEFAULT_AUTO_FIELD setting or the RegistrationConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
```

- _Resolution_
    Added `DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'` to the` base.py` settings file. This setting ensures that any auto-created primary keys use BigAutoField by default, avoiding potential issues with primary key type mismatches in the future.
- _Code:_
```
  # in base.py
DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'
```

3. Ensured Existence of Static Directory:

-  _Warning Fixed:_
`?: (staticfiles.W004) The directory '/home/snipher/Open Source Contributions/pingprivately/privateping/PrivatePing/../assets/static' in the STATICFILES_DIRS setting does not exist.
`
- _Resolution:_
Created the missing static directory under the assets directory. This change ensures that Django’s static file management can find the specified directory, preventing any issues during static file collection and serving.

**Impact:**

- **Improved Configuration:** Ensuring `DEFAULT_AUTO_FIELD` is set improves consistency and future-proofs the database schema management.
- **Resolved Static File Warning:** Creating the static directory as specified in` STATICFILES_DIRS` prevents runtime errors related to static file handling.
- **Clean Development Logs:** By addressing these warnings, the development server logs are cleaner, making it easier for developers to spot and address other potential issues.

**Notes:**

- Developers are encouraged to verify that their local settings align with these changes, especially if they have custom configurations.
- Further improvements and suggestions are welcome to continue enhancing the project configuration and adherence to best practices.
- Refactored registration templates to inherit from the base template using Jinja, reducing repetition and improving maintainability.
- Refactored code to use BASE_DIR for static file paths instead of os for better readability and consistency.

**Summary:**

This pull request addresses critical configuration warnings in the Django project, improving the development experience and ensuring adherence to best practices. Please review the changes and merge them if they meet the project standards.

### CODE REVIEW

**Comment by princekhunt** [comment]
@snipher-marube Thanks for the pr.

On it...

**Comment by snipher-marube** [comment]
> @snipher-marube Thanks for the pr.
> 
> On it...

Alright Thank you

**Review by princekhunt — COMMENTED** [review]
in second commit, 
whitenoise is already added, so no need to add again.

File: PrivatePing/settings/base.py [hunk 1]
```diff
@@ -95,4 +95,6 @@
     os.path.join(BASE_DIR, '../assets/static')
 ]
 STATIC_ROOT = os.path.join(BASE_DIR, '../assets/')
-STATIC_URL = '/static/'
\ No newline at end of file
+STATIC_URL = '/static/'
+
+DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'
```
Comment (princekhunt) [inline PrivatePing/settings/base.py:107]: I find only this one important.
no need to change anything in assets.
if you think, changes in assets are necessary, kindly explain.

Comment (snipher-marube) [reply PrivatePing/settings/base.py:107]: Yes yes, I overlooked the importance of WhiteNoise in my last commit. I have reverted back the assets but only restructured the directories to ensure proper static file handling, where I separated the staticfiles from static

**Comment by princekhunt** [comment]
Continuing updation in [princekhunt:dev ](https://github.com/princekhunt/privateping/tree/dev)

**Pull request merged**